components/FormToCreateShoppingItem.js

```diff
@@ -0,0 +1,108 @@
+import styled from "styled-components";
+
+export default function FormToCreateShoppingItem({ onAddItem, categories }) {
+  console.log("Categories: ", categories);
+
+  function handleSubmit(event) {
+    event.preventDefault();
+    const formData = new FormData(event.target);
+    const newItem = Object.fromEntries(formData);
+    newItem.quantity = Number(newItem.quantity);
+    newItem.imageUrl = "placeholder_1.png";
+
+    onAddItem(newItem);
+    event.target.reset();
+  }
+
+  return (
+    <article>
+      <container>
+        <h1>Add an item to the list:</h1>
+        <form onSubmit={handleSubmit} data-js="form">
+          <StyledFieldset>
+            <StyledLabel>
+              new shopping item*:
+              <StyledInput
+                name="name"
+                type="text"
+                required
+                placeholder="e.g., salt"
+              />
+            </StyledLabel>
+            <StyledLabel>
+              number*:
+              <StyledInput name="quantity" type="number" required />
+            </StyledLabel>
+            <StyledLabel>
+              category*:
+              <select
+                key="category"
+                name="category"
+                data-js="category"
+                required
+              >
+                <option value="">please select a category</option>
+                {categories.map((category) => (
+                  <option key={category.name} value={category.name}>
+                    {category.name}
+                  </option>
+                ))}
+                ;
+              </select>
+            </StyledLabel>
+            <StyledLabel>
+              comment:
+              <StyledInput
+                name="comment"
+                type="text"
+                placeholder="Enter comments here..."
+              />
+            </StyledLabel>
+            <StyledNote>
+              Required fields are followed by{" "}
+              <span aria-label="required">*</span>.
+            </StyledNote>
+            <StyledButton type="submit">Submit</StyledButton>
```

Comment:
a button is "submit" by default so you don't have to write the type here

Reply:
Right. Thanks for pointing this out.